### DISCUSSION TITLE
Dangling symlinks preventing deployment

### DISCUSSION THREAD

**Discussion by abingham** [heroku/buildpacks#42] [General]
This probably isn't a buildpack issue per se, but I don't know where else to start. It appears to be a problem (bug?) in codon, but I can't find a forum for that.

In short: I replaced a symlink in my repo with a normal file, and now my deployments fail with the error: 
```
cp: not writing through dangling symlink '/tmp/codon/tmp/cache/.heroku/requirements.txt'
 !     Push rejected, failed to compile Python app.
 !     Push failed
```

`requirements.txt` is the file that used to be a symlink and is now a file. Assuming this isn't the place to really deal with this issue, where can I go? Or if this is the right place, what do I do?

**Comment by abingham** [comment]
The (or at least an) answer seems to [clearing the build cache](https://help.heroku.com/18PI5RSY/how-do-i-clear-the-build-cache). This is pretty clearly a bug somewhere, but clearing the cache lets me work past it.

**Comment by edmorley** [comment]
@abingham Hi! This repository is for Heroku's next-generation Cloud Native Buildpacks (CNBs), which are currently in preview and not yet usable directly on Heroku. From the error message, this appears to be an issue with the existing generation buildpacks instead, and specifically with the Python buildpack. For issues like this I would either open a support ticket (https://help.heroku.com) or file an issue here:
https://github.com/heroku/heroku-buildpack-python

**Reply by edmorley** [reply]
I've filed this as https://github.com/heroku/heroku-buildpack-python/issues/1675 and have a fix in progress :-)
